crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -1883,9 +1883,76 @@ impl TwoQubitBasisDecomposer {
     }
 }
 
+static MAGIC: [[Complex64; 4]; 4] = [
+    [
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+    ],
+    [
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+    ],
+    [
+        C_ZERO,
+        c64(0., FRAC_1_SQRT_2),
+        c64(-FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+    ],
+    [
+        c64(FRAC_1_SQRT_2, 0.),
+        C_ZERO,
+        C_ZERO,
+        c64(0., -FRAC_1_SQRT_2),
+    ],
+];
+
+#[pyfunction]
+pub fn two_qubit_local_invariants(unitary: PyReadonlyArray2<Complex64>) -> [f64; 3] {
+    let mat = unitary.as_array();
+    // Transform to bell basis
+    let bell_basis_unitary = transpose_conjugate(aview2(&MAGIC)).dot(&mat.dot(&aview2(&MAGIC)));
```

Comment:
There's possibly a speedup and accuracy improvement to be had by storing the adjoint statically, and doing the asymmetric non-normalised trick we do in the other Weyl code.

Reply:
Added the static in: https://github.com/Qiskit/qiskit/pull/12739/commits/c4e369aad539e80b2264d9cabcf06fafcbd771fd